cli/src/cmd/utils.rs

```diff
@@ -156,28 +163,66 @@ pub fn needs_setup(abi: &Abi) -> bool {
 
 pub fn unwrap_contracts(
     contracts: &BTreeMap<ArtifactId, ContractBytecodeSome>,
+    deployed_code: bool,
 ) -> BTreeMap<ArtifactId, (Abi, Vec<u8>)> {
     contracts
         .iter()
         .map(|(id, c)| {
-            (
-                id.clone(),
-                (
-                    c.abi.clone(),
-                    c.deployed_bytecode.clone().into_bytes().expect("not bytecode").to_vec(),
-                ),
-            )
+            let bytecode = if deployed_code {
+                c.deployed_bytecode.clone().into_bytes().expect("not bytecode").to_vec()
+            } else {
+                c.bytecode.clone().object.into_bytes().expect("not bytecode").to_vec()
+            };
+
+            (id.clone(), (c.abi.clone(), bytecode))
         })
         .collect()
 }
 
+/// Data struct to help `ScriptSequence` verify contracts on `etherscan`.
+pub struct VerifyBundle {
+    pub num_of_optimizations: Option<usize>,
+    pub known_contracts: BTreeMap<ArtifactId, (Abi, Vec<u8>)>,
+    pub etherscan_key: Option<String>,
+    pub project_paths: ProjectPathsArgs,
+}
```

Comment:
I'll leave this for a followup to move all script/verify related code from utils to their own module(s)

Reply:
Wait, the end point of  APi  not been completed